File: TravellerCreatorModels/RTTWorldgen/RTTWorldgenSector.cs
using TravellerCreatorModels.Enums;
using TravellerCreatorModels.Interfaces;

namespace TravellerCreatorModels.RTTWorldgen;

public class RTTWorldgenSector : ISector
{
    public SectorType SectorType => SectorType.RTTWorldgen;
    public List<ISubsector> Subsectors { get; set; } = new();
}

File: TravellerCreator/ClientApp/src/app/sector-creator/models/rtt-worldgen/rtt-worldgen.sector.ts
import {ISector} from "../interfaces/sector";
import {SectorType} from "../enums/sector-type";
import {RttWorldgenSubsector} from "./rtt-worldgen.subsector";

export class RttWorldgenSector implements ISector {
  sectorType: SectorType;
  subsectors: RttWorldgenSubsector[];
}
